File: Problems/Parking Lot/Java/payments/TruckPaymentStrategy.java
package payments;

import java.time.Duration;

import ticket.Ticket;

public class TruckPaymentStrategy implements Payments {
    private static final double TRUCK_RATE_PER_HOUR = 5.0; // Example rate per hour for truck parking

    @Override
    public double pay(Ticket ticket) {
        Duration duration = Duration.between(ticket.getEntryTime(), ticket.getExitTime());

        double parkedHours = Math.ceil(duration.toSeconds() / 3600.0);

        return parkedHours * TRUCK_RATE_PER_HOUR;
    }
    
}

File: Problems/Parking Lot/Golang/payments/truck_payment_strategy.go
package payments

import (
	"math"
	"parking-lot/ticket"
)

type TruckPaymentStrategy struct{}

const (
	TRUCK_RATE_PER_HOUR = 3.0
)

func (tps *TruckPaymentStrategy) pay(ticket ticket.Ticket) float64 {
	duration := ticket.GetExitTime().Sub(ticket.GetEntryTime())

	parkedHours := math.Ceil(duration.Seconds() / 3600.0)

	return parkedHours * TRUCK_RATE_PER_HOUR
}
